packages/carbon-graphs/src/js/controls/PairedResult/helpers/helpers.js

```diff
@@ -409,6 +409,23 @@ const processDataPoints = (graphConfig, dataTarget, reflow = false) => {
       values: [],
     },
   };
+
+  if (!reflow && dataTarget.values.length < 1) {
+    const types = ['high', 'mid', 'low'];
+    types.forEach((t) => {
+      const label = getValue(dataTarget.label, t);
+      if (label && label.display) {
+        if (
+          !utils.hasValue(
+            graphConfig.shownTargets,
+            `${dataTarget.key}_${t}`,
+          )
+        ) {
+          graphConfig.shownTargets.push(`${dataTarget.key}_${t}`);
+        }
+      }
+    });
+  }
```

Comment:
here's a high level suggestion to be more declarative and have a more functional mindset, which results in easier to read code with less side effects. I extracted the logic into its own helper that accepts the necessary inputs, converted the forEach + push into a filter call, and made use of the spread operator to combine the final array. The spread will need to be changed if graphConfig is declared with a `const`. let me know what you think.
```js
// util defined outside of processDataPoints. needs a better name
const geValues = (reflow, dataTarget, label) => {
  if (!reflow && !dataTarget.values.length < 1) {
      const types = ['high', 'mid', 'low'];
      const shownTargets = types.filter(type => {
        const label = getValue(dataTarget.label, type);
        const hasValue = utils.hasValue(graphConfig.shownTargets, `${dataTarget.key}_${type}`)
        if (label && label.display && !hasValue) {
            return `${dataTarget.key}_${type}`;
          }
        }
      });

    return shownTargets;
  }
  return [];
}
 
const shownTargets = getValues(reflow, dataTarget, label);
graphConfig.showTargets = [...graphConfig.showTargets, ...shownTargets];
```

Reply:
@yuderekyu Your idea sounds good, but [processDataPoint](https://github.com/cerner/terra-graphs/pull/111/files#diff-d04c4cf6f73329ea29bf2a081e4f7b5d5ddf52580b58d6ce58bf9c7b2728fb1eR385) method is used to initialize some variables before generating a graph content. Showntargets is one of the variables that need to be initialized in that method, so to keep everything organized I just added this content in the method where it needs to be executed. And also it is used only in one place so I felt it is not necessary to have separate util. Correct me if i am wrong.

Reply:
sure, i'm just afraid that processDataPoint is doing way too much - splitting the logic into smaller transforms makes it easier to reason about, and less coupled - i think the whole function could make do with a re-work, which is probably out of scope of this. I do think it's best to leave the code better than what we left it, and here, we're adding another layer of complexity for this function. In this case, splitting the logic into separate function is more about being modular, and less about reuse

Reply:
We do have a future project to do a code overhaul for a major update that also cleans up functions and streamlines the logic. So I would suggest staying within the scope of the bug-fix for now and make a note of these suggestions for the next project.

Reply:
I'd argue that making code lean at the outset saves time and effort in the long run. You've introduced new logic here, why not spend the time to make it encapsulated, coding around the complex legacy code that's here? I won't stop you from merging this, but just food for thought. More often than not, "later" becomes "never" 🤷🏼

Reply:
+1 to @yuderekyu  suggestion (assuming it doesn't blow up the scope too much. I like it when we can try and modularize where we can and I do think processDataPoint is a little busy as it is.

Reply:
Corrected [here](https://github.com/cerner/terra-graphs/pull/111/commits/a4eabe34b7595d302599c25f50a034acf6d5550b#diff-d04c4cf6f73329ea29bf2a081e4f7b5d5ddf52580b58d6ce58bf9c7b2728fb1eR383) as needed.